src/ray/raylet/worker_pool.cc

```diff
@@ -979,35 +973,13 @@ void WorkerPool::PushWorker(const std::shared_ptr<WorkerInterface> &worker) {
   bool found;
   bool used;
   TaskID task_id;
-  InvokePopWorkerCallbackForProcess(state.starting_dedicated_workers_to_tasks,
-                                    worker->GetStartupToken(),
-                                    worker,
-                                    PopWorkerStatus::OK,
-                                    &found,
-                                    &used,
-                                    &task_id);
-  if (found) {
-    // The worker is used for the actor creation task with dynamic options.
-    if (!used) {
-      // Put it into idle dedicated worker pool.
-      // TODO(SongGuyang): This worker will not be used forever. We should kill it.
-      state.idle_dedicated_workers[task_id] = worker;
-    }
-    // We either have an idle worker or a slot to start a new worker.
-    if (worker->GetWorkerType() == rpc::WorkerType::WORKER) {
-      TryPendingPopWorkerRequests(worker->GetLanguage());
-    }
-    return;
-  }
-
   InvokePopWorkerCallbackForProcess(state.starting_workers_to_tasks,
                                     worker->GetStartupToken(),
                                     worker,
                                     PopWorkerStatus::OK,
                                     &found,
                                     &used,
                                     &task_id);
-  // The worker is not used for the actor creation task with dynamic options.
   if (!used) {
     // Put the worker to the idle pool.
```

Comment:
@architkulkarni @iycheng should we check if the worker is an actor and used, and terminate the worker if it's already used for an actor?

Reply:
Not sure about this, they must currently already be being terminated somewhere else right? (Are you suggesting we move that logic here?)

Reply:
yes we probably should ensure actor worker is not reused here.